src/librustc/traits/project.rs

```diff
@@ -1247,6 +1247,34 @@ fn confirm_callable_candidate<'cx, 'gcx, 'tcx>(
                                               fn_sig,
                                               flag);
 
+    if selcx.projection_mode().is_any() {
```

Comment:
Hmm. This is somehow hackier than I had imagined. What I had hoped was that we would attach a (generic) filter function to the `FulfillmentContext` -- when not in trans, this filter function would accept all work, but within trans, it could be more selective.